test/puppetlabs/stockpile/queue_test.clj

```diff
@@ -101,18 +102,32 @@
   (call-with-temp-dir-path
    (fn [tmpdir]
      (let [qdir (.toFile (.resolve tmpdir "queue"))
-           newq (stock/create qdir)]
-       (let [entry-1 (store-str newq "foo" "meta foo")
-             q (stock/open qdir)
-             read-entries (stock/reduce q conj ())]
-         (is (= [entry-1] read-entries))
-         (is (= "foo" (slurp-entry q entry-1)))
-         (let [entry-2 (store-str q "bar" "meta bar")]
-           (let [q (stock/open qdir)
-                 read-entries (stock/reduce q conj #{})]
-             (is (= #{entry-2 entry-1} read-entries))
-             (is (= "foo" (slurp-entry q entry-1)))
-             (is (= "bar" (slurp-entry q entry-2))))))))))
+           ent-1-id (let [q (stock/create qdir)
```

Comment:
It seems like the complexity of this test is rooted in trying to test two separate things in one test. If I'm understanding this correctly, ent-1-id is testing the storage of a stream without a metadata-string and ent2-id includes a metadata string. Seems like it would make the tests simpler to have them be separate (i.e. two separate testing clauses or two different deftests). Is there a reason to keep them together?